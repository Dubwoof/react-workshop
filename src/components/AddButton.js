import React from "react";

export const AddButton = ({ func }) => <button onClick={func}>test</button>;
