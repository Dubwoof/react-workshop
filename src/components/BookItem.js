import React from "react";
import { Link } from "react-router-dom";

export const BookItem = ({ item, index }) => {
  return (
    <Link to={`/book/${item.isbn}`}>
      <div>
        {index + 1} {item.title}
      </div>
    </Link>
  );
};
