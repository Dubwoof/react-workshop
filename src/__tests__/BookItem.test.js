import React from "react";
import { BookItem } from "../components/BookItem";
import ReactTestRenderer from "react-test-renderer";
import { MemoryRouter as Router } from "react-router-dom";

const book = { title: "My Title" };

describe("A BookListItem component", function() {
  it("renders a book item correctly", () => {
    const tree = ReactTestRenderer.create(
      <Router>
        <BookItem item={book} />
      </Router>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
