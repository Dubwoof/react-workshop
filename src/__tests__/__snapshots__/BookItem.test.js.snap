// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`A BookListItem component renders a book item correctly 1`] = `
Array [
  <a
    href="/book/undefined"
    onClick={[Function]}
  >
    <div>
      NaN
       
      My Title
    </div>
  </a>,
  <a
    className="noDecoration"
    href="/book/undefined/edit"
    onClick={[Function]}
  >
    âœŽ
  </a>,
]
`;
